doc/vxlan/EVPN/EVPN_VxLAN_Multihoming.md

```diff
@@ -0,0 +1,1450 @@
+
+**Note:** DF election timer value is set to 3 sec as per RFC 7432, and is not configurable.
+
+**Note:** As per the standard, if two routers do not agree on a given DF election algorithm, they all should fallback on default modulo based scheme proposed in RFC 7432. This fallback option is not supported and SONiC device will not own the DF role in such interop cases.
+
+### 2.2.2.2 Interoperability between ES Multihoming VTEPs
+There are two types of interoperabilities in EVPN MH network:
+- (a) Interoperability between VTEPs which are multihoming one or more common ES
+- (b) Interoperability between VTEPs which are not multihoming any common ES
+
+SONiC will support interoperability (b). 
+
+SONiC will not be able to support interoperability (a) if the EVPN MH participating VTEP does not support highest preference based DF election algorithm.
+It is recommended that all of the VTEPs multihoming one or more common ES are the SONiC devices.
+
+### 2.2.3 Split Horizon Filtering
+Split Horizon filtering is achieved by identifying the set of VTEPs Multihoming a given Ethernet Segment. 
+
+Using AD-per-ES routes, FRR constructs list of participating VTEPs for each ESI. And Fpmsyncd installs EVPN_SPLIT_HORIZON_TABLE entry in APP-DB that contains list of remote VTEPs multihoming the given PortChannel interface.
+
+The Isolation group SAI object used for achieving Local Bias and Split Horizon Filtering in the MCLAG will be reused for EVPN MH. In EVPN MH an Isolation group for every Peer Multihoming VTEP is created as opposed to a single group for MCLAG. The Isolation group members are populated based on the EVPN_SPLIT_HORIZON_TABLE APP-DB entry. Traffic received from the Tunnel towards the peer multihoming VTEP will not be forwarded to the members of the Isolation group.
+
+### 2.2.4 Linux Kernel support for EVPN Multihoming
+EVPN Multihoming feature requires L2 next-hop group (NHG) support in the Linux kernel that is available in v5.10.
+EVPN Multihoming feature will not be supported in SONiC releases running 4.x kernel versions.
+
+#### 2.2.4.1 L2 Next-hop Groups
+Remote MAC addresses received via BGP EVPN Type-2 route updates having non-zero ESI will be installed in the Linux kernel fdb with L2 nhid as show below:
+```
+# bridge fdb show
+00:00:0a:0b:00:03 dev vtep1-100 vlan 100 extern_learn master Bridge
+00:00:0a:0b:00:03 dev vtep1-100 nhid 536870913 self extern_learn
+```
+
+The kernel nexthop table maintains the next-hop groups and member entries.
+```
+# ip nexthop show
+id 268435458 via 1.1.1.1 scope link fdb
+id 268435459 via 2.2.2.2 scope link fdb
+id 536870913 group 268435458/268435459 fdb
+```
+In the example output, L2 NHID group 536870913 has two member nexthops - 268435458 and 268435459 - which resolve to single path VTEP nexthops.
+
+
+#### 2.2.4.2 BUM traffic handling
```

Comment:
Which component is  responsible for installing tc rules to kernel?

Reply:
Ideally it should be FRR (zebra.)